File: app/src/androidTest/java/it/droidcon/testingdaggerrxjava/test2/TestUserInteractorModule.java
package it.droidcon.testingdaggerrxjava.test2;

import dagger.Module;
import dagger.Provides;
import it.droidcon.testingdaggerrxjava.core.UserInteractor;
import javax.inject.Singleton;
import org.mockito.Mockito;

@Module
public class TestUserInteractorModule {
  @Provides @Singleton
  public UserInteractor provideUserInteractor() {
    return Mockito.mock(UserInteractor.class);
  }
}

File: app/src/androidTest/java/it/droidcon/testingdaggerrxjava/test2/TestUserInteractorModule.kt
package it.droidcon.testingdaggerrxjava.test2

import com.nhaarman.mockito_kotlin.mock
import dagger.Module
import dagger.Provides
import it.droidcon.testingdaggerrxjava.core.UserInteractor
import javax.inject.Singleton

@Module class TestUserInteractorModule {
    @Provides @Singleton fun provideUserInteractor(): UserInteractor = mock()
}
